frontend/app/src/services/history/history-api.ts

```diff
@@ -58,34 +55,45 @@ export class HistoryApi {
     return this.axios
       .get<ActionResult<TradeLocation[]>>('/locations/associated', {
         validateStatus: validStatus,
-        transformResponse: this.responseTransformer
+        transformResponse: basicAxiosTransformer
       })
       .then(handleResponse);
   }
 
-  async trades(
-    location?: TradeLocation,
-    onlyCache?: boolean
-  ): Promise<PendingTask> {
-    const params = {
-      asyncQuery: true,
-      onlyCache: onlyCache ? onlyCache : undefined,
-      location
-    };
+  async internalTrades<T>(
+    payload: TradeRequestPayload,
+    async: boolean
+  ): Promise<T> {
     return this.axios
-      .get<ActionResult<PendingTask>>('/trades', {
-        params: axiosSnakeCaseTransformer(params),
-        validateStatus: validStatus,
-        transformResponse: this.responseTransformer
+      .get<ActionResult<T>>('/trades', {
+        params: axiosSnakeCaseTransformer({
+          asyncQuery: async,
+          ...payload
+        }),
+        validateStatus: validWithParamsSessionAndExternalService,
+        transformResponse: tradeAxiosTransformer
       })
       .then(handleResponse);
   }
 
+  async tradesTask(payload: TradeRequestPayload): Promise<PendingTask> {
+    return this.internalTrades<PendingTask>(payload, true);
+  }
+
+  async trades(
```

Comment:
Do you think it would be too much of an effort to introduce strict validation here with a new zod schema?

Reply:
Ooo, alright, I will see is it easy to do or not

Reply:
The parsing functionality is added. 
But somehow I notice the trades section to be slower.
But the transactions section is not that slow, I think I need to investigate more... @kelsos

Reply:
And I also having some trouble with ts typing for this part of the zod `frontend/app/src/services/history/types.ts`, I leave it with `any` for now.

Reply:
> The parsing functionality is added. But somehow I notice the trades section to be slower. But the transactions section is not that slow, I think I need to investigate more... @kelsos

I've found that not the zod parsing that make it slow

Reply:
The async_query=true is slow, but the only_cache should not be